server/lib/genome/importers/api_importers/go/importer.rb

```diff
@@ -36,13 +36,13 @@ def create_gene_claims
       categories.each do |category, go_id|
         start = 0
         rows = 500
-        genes = api_client.genes_for_go_id(go_id, start, rows)
-        while genes.count.positive? do
+        loop do
+          genes = api_client.genes_for_go_id(go_id, start, rows)
           genes.each do |gene|
             create_gene_claim_for_entry(gene, category) if gene['taxon_label'] == 'Homo sapiens'
           end
+          break if genes.count < rows
```

Comment:
Does this break if the number of rows on the last page happens to be the same as the page size and run into the `raise StandardError` in the api client on the next request?

If so, maybe you want to handle that separately? 

You can directly check the http response code for 404 (as it is currently doing with 200), or you could match on the response type like this example [here](https://ruby-doc.org/3.3.5/stdlibs/net/Net/HTTP.html#class-Net::HTTP-label-Following+Redirection) if you find that to be cleaner. (Note it will not be redirection as the subclass but NotFound). I could be misunderstanding what's going on though.

Reply:
yeah, totally. I was being lazy.

it seems like the GO API returns the same response payload for a bunch of different classes of errors:

```json
{"detail":"Item with ID GO:0008233 not found"}
```

You get a variation of this ^ if you [request a nonexistent entity](https://api.geneontology.org/api/bioentity/function/GO:0008233sdflkjasdf), but also if you [make a request where "start" is beyond the number of records](https://api.geneontology.org/api/bioentity/function/GO:0008233?start=9000&rows=500). I was trying to avoid catching 404s because, like, those are nice to know about for other reasons, but maybe we can condition this such that we raise the exception if the initial request with `start=0` returns a 404, but we just break the loop once we're starting from a higher number.

Reply:
There's also a good chance that this is a recent bug that will get fixed, but the issue tracker for the GO site has like 50 open PRs and hundreds of open issues, so... idk